File: SISTEMAS_CLASES/CarritoDeCompras.java
package SISTEMAS_CLASES;

public class CarritoDeCompras {
    Producto ProductoSeleccionado;
    int Cantidad;
    int Total;

    public CarritoDeCompras(int cantidad,int total){

        this.Cantidad=cantidad;
        this.Total=total;
    }
    @Override
    public String toString() {
        return "CarritoDeCompras{" +
                "ProductoSeleccionado=" + ProductoSeleccionado +
                ", Cantidad=" + Cantidad +
                ", Total=" + Total +
                '}';
    }

    public Producto getProductoSeleccionado() {
        return ProductoSeleccionado;
    }

    public int getCantidad() {
        return Cantidad;
    }

    public int getTotal() {
        return Total;
    }
}

File: SISTEMAS_CLASES/CarritoDeCompras.py

import re
from Producto import Producto
class CarritoDeCompras:
    
    productoSeleccionado:Producto
    Cantidad=int
    TotalAPagar=int

    def __init__(self,producto:Producto):
        self.roductoSeleccionado=producto
        self.Cantidad=1
    

    def AumentarCantidad(self):
        cantidad= int (input("cuantos desea agregar: "))
        self.Cantidad=self.Cantidad +cantidad

    def DisminuirCantidad(self):
        cantidad= int (input("cuantos desea eliminar: "))
        self.Cantidad=self.Cantidad +cantidad

    def CalcularTotalProducto(self): 
        print (self.Cantidad)
        print(self.productoSeleccionado.Precio)
        self.TotalAPagar=self.Cantidad* self.productoSeleccionado.Precio
    
        return print("regreso")


    def ProductoSeleccionado(self):
        return self.productoSeleccionado
    
    def Catidad(self):
        return self.Cantidad
    
    def ToTalAPagar(self):
        return self.TotalAPagar
        
        


    



    